cornflow/shared/authentication.py

```diff
@@ -76,10 +80,30 @@ def get_user_obj_from_header(headers):
         data = Auth.decode_token(token)
         user_id = data["user_id"]
         user = UserModel.get_one_user(user_id)
-        if not user:
+        if user is None:
             raise ObjectDoesNotExist("User does not exist, invalid token")
         return user
 
+    @staticmethod
+    def get_permission_for_request(req, user_id):
+        method, url = Auth.get_request_info(req)
+        user_role = UserRoleModel.get_one_user(user_id=user_id)
+        if user_role is None:
+            raise NoPermission("You do not have permission to access this endpoint")
+
+        action_id = PERMISSION_METHOD_MAP[method]
+        view_id = ApiViewModel.query.filter_by(url_rule=url).first().id
+
+        for role in user_role:
+            has_permission = PermissionViewRoleModel.get_permission(
+                role.role_id, view_id, action_id
+            )
+
+            if has_permission is not None:
+                return True
```

Comment:
shouldn't PermissionViewRoleModel return True / False?

Reply:
I can change it.